js/core/utils/dom.js

```diff
@@ -152,3 +152,11 @@ export const replaceWith = (element, newElement) => {
 
     return newElement;
 };
+
+export const isElementInDom = $element => {
+    const element = $element?.get(0);
+    const shadowHost = element?.getRootNode().host;
+    const elementOrShadowHost = shadowHost || element;
+
+    return !!$(elementOrShadowHost).closest(getWindow().document).length;
```

Comment:
this variable looks strange for me - its name repeats the operation.
```suggestion
    return !!$(shadowHost || element).closest(getWindow().document).length;
```